### PR TITLE
Move to typescript from js

### PR DESCRIPTION
**Pull request by anujdeshpande**

### CODE REVIEW

**Comment by rcooke-warwick** [comment]
@anujdeshpande just made some changes to get the TS version working - a couple of things to consider are:

1. do we zip the entire contents of the output folder, or only files of a certain type
2. we need a spec.json or similar (env.example style thing) for PCB type projects - at the moment were checking against a json schema but we can change format if we want

**Comment by rcooke-warwick** [comment]
reminder - if we want to deploy anything here make sure we don't need to alter balenaci/scripts

**Comment by anujdeshpande** [comment]
@rcooke-warwick awesome!
About 1 - I think yes. The outputs folder has files meant for the manufacturing folks. CI shouldn't have any logic on determining if a file feels out of place there. 

About 2 - Does subcategories in repo.yml make sense instead

**Comment by rcooke-warwick** [comment]
@anujdeshpande 
an example of a repo.yaml:

type: hardware
release: github
parameters:
    - thickness: 2mm
    - finish: smooth
    - material: copper

Something like this? A couple of questions I have are:
1. I don't know much about YAML, is it as simple as json to validate/filter vs a schema
2. Is doing that invasive and against the point of repo.yaml, which I've got a feeling should only have data about the repo, not info that is only relevant to a single PR/manufacturing thing.

We could try a different yaml, like spec.yaml or manufacture.yaml, I suppose its a lot more readable and easier to edit than a json file

**Comment by rcooke-warwick** [comment]
Another thing I jsut thought of - as we're just grabbing the contents of the output folder, we can basically deploy our email loop to any projects just by:

1. adding 'type:hardware' to the repo.yaml
2. dropping in a spec.json/ whatever we decide to carry the other parameters like thickness/material etc
3. creating an outputs folder, where eveythig that is used for manufacture can just be dropped in to, and that gets added to the release

File: src/index.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+		console.log(fileName.name);
+		// if its a directory, do it again - if its a file, zip it
+		if(fileName.isDirectory()){
+			console.log('this is a directory')
+			packageFiles(Path.join(path, fileName.name));
+		} else {
+			console.log('this is a file')
+			const fileRead = fs.readFileSync(Path.join(path, fileName.name))
+			const fileContent = Buffer.from(fileRead).toString('utf8');
+			zip.file(fileName.name, fileContent) // what should the files get called? is it ok just to dump all the files in there?
```
Comment (anujdeshpande) [inline src/index.ts:49]: @rcooke-warwick Should we take this as an input from capitano? I think it makes sense to accept this as branch from which release is created or even better the subject of the email.

File: src/index.ts [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+	packageFiles(Path.join(params.folder, 'outputs'));
+
+	const data = await zip.generateAsync({
+		type: 'nodebuffer',
+		compression: 'DEFLATE'
+	});
+	fs.writeFileSync(Path.join(options.output, `release.zip`), data, 'binary');
+};
+
+// check the spec file against schema for the project type - this will have to change if we get rid of the spec.json file from the repos
```
Comment (anujdeshpande) [inline src/index.ts]: @rcooke-warwick I can't remember, but have we discussed doing something like this (removing the spec.json)? Can't seem to think of why we'd want that

File: src/index.ts (hunk 1 above)
Comment (rcooke-warwick) [reply src/index.ts:49]: @anujdeshpande if/when we moved it into versionbot then I agree, it will be taken from the cloned branch.

Comment (anujdeshpande) [reply src/index.ts:49]: `Moving this to versionbot` seems like something we can discuss in today's call then

File: src/index.ts (hunk 2 above)
Comment (rcooke-warwick) [reply src/index.ts]: @anujdeshpande at the moment we were using spec.json to hold parameters not in the source file, like thickness/material. We talked about maybe having a file per manufacturing run instead though but we didn't decide on anything

**Comment by anujdeshpande** [comment]
@rcooke-warwick can we merge this?

**Comment by rcooke-warwick** [comment]
> @rcooke-warwick can we merge this?

almost @anujdeshpande, it was failing the tests so I just pushed a commit that will hopefully fix it

**Comment by rcooke-warwick** [comment]
@nazrhom whats up with these checks, the store-npm has been stuck for ages

**Comment by anujdeshpande** [comment]
hey @nazrhom can you please review this?

**Review by rcooke-warwick — APPROVED** [review]

**Pull request merged**